### ISSUE TITLE
prefer-const false positive on array destructuring

### ISSUE THREAD

**Issue by Wilfred** [eslint/eslint#6926]
Given the code:

```js
function foo() {
  let [x, y] = global.someFunc();
  x = x + 1;
}
```

prefer-const complains:

```
$ eslint --no-eslintrc --rule 'prefer-const: "warn"' --parser-options '{ecmaVersion: 6}' foo.js
foo.js
  2:11  warning  'y' is never reassigned. Use 'const' instead  prefer-const
```

I think this is a false positive. I can't use `const` in this case because I'm reassigning `x`.

**Comment by eslintbot** [comment]
Hi @Wilfred, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by Wilfred** [comment]
I'm using eslint 3.2.2. Let me know if you need any further information.

**Comment by gyandeeps** [comment]
You wanna use `"destructuring": "all"` option.
Docs: http://eslint.org/docs/rules/prefer-const#options

**Comment by Wilfred** [comment]
Thanks, I missed that.

Is there any way of rewriting my code to satisfy `destructuring: "any"`? The best I can see is:

```js
const x_and_y = global.someFunc();
let x = x_and_y[0];
const y = x_and_y[1];
```

but that's significantly more verbose. I'm trying to understand the use case for `destructuring: "any"`.

**Comment by mysticatea** [comment]
For example:

```js
const [x, y] = someFunc();
const modifiedX = x + 1;
```

**Comment by Wilfred** [comment]
Aha, makes sense. In which case, I don't think there's anything actionable here.
